docs/code-review.md

```diff
@@ -0,0 +1,330 @@
+      - [Don't Panic](#dont-panic)
+      - [Discarded errors](#discarded-errors)
+      - [Adding context](#adding-context)
+      - [Indent Error Flow](#indent-error-flow)
+    - [Naming convention](#naming-convention)
+      - [Variable Names](#variable-names)
+      - [Receiver's name](#receivers-name)
+      - [Be consistent](#be-consistent)
+      - [Search for an element](#search-for-an-element)
+
+## What to look for in a code review?
+
+### Code review checklist
+
+Here is a summary of the things every reviewer should be aware of while doing a CLI code review:
+
+- [ ] Does the PR have unit tests/e2e tests that covers all the scenarios?
+  *We should look forward to have all scenarios covered.*
+- [ ] Does the PR description explain what it does/solves?
+  *In the future we may need to know why the PR was created and what issues it solved, so this should be made clear in the description of the PR.*
+- [ ] Does it affect other services (actions/vscode plugin/pipeline/graphql/json logs)?
+  *It can affect other services that are not the CLI itself and break scenarios that are not contemplated on the CLI repository, so we need to bear in mind all those services when reviewing a PR*
+- [ ] Does it need to add analytics?
+  *We need to add analytics to know the adoption of new features and how the users use the product*
+- [ ] Are there any code smells?
+  we should develop clean code so that if a new developer starts working on the project, he/she can understand the code as soon as possible*
+
+### PR Description
+
+The description of the pull request should state what issue it solves (or explain if there is no associated issue) and how it is solved.
+
+If required, the author could be asked to provide screenshots or videos explaining what was wrong and how it was fixed.
+
+### PR comments
+
+Comments are the basis of all pull requests. In these the reviewers will ask questions about the code, ask for changes, split pull requests into several, discuss what is the best approach to improve the CLI code.
+
+A respectful tone will be used adding whenever necessary documentation to reinforce the comments made.
+
+### Tests
+
+All new features or bug fixes should have at least unit tests that cover the use case. If for any reason unit tests cannot be created, at least one e2e test should be created to prove that the new functionality works correctly.
+
+Tests should fail with helpful messages saying what was wrong, with what inputs, what was actually got, and what was expected. Assume that the person debugging your failing test is not you, and is not your team.
+
+#### Unit tests
+
+The code should be testable by unit functions, which means that each function should perform an action and this action should be testable.
+
+Only one test will be created per function in which it will be tested by table driven tests with test names descriptive enough for any user to know what it does just by reading the name of the test. For example:
```

Comment:
I wouldn't enforce the fact of creating just one test function and use table driven tests. Some scenarios, specially the ones which verify errors or normal output (like the example you posted below) force you to have some kind of flag and include an `if` within the test to do different asserts. In general, that is a bad practice in testing